collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java

```diff
@@ -565,117 +684,83 @@ else if (columnNames.contains(path)) { // if column present in sheet
-
-        // Find an album with a title matching the form_id
-        // Technically there could be multiple albums that match
-        // We just use the first one that matches
-        if (feed.albums != null) {
-            for (AlbumEntry album : feed.albums) {
-                if (jrFormId.equals(album.title)) {
-                    albumToUse = album;
-                    break;
-                }
-            }
-        }
 
-        // no album exited, so create one
-        if (albumToUse == null) {
-            AlbumEntry newAlbum = new AlbumEntry();
-            newAlbum.access = "private";
-            newAlbum.title = jrFormId;
-            newAlbum.summary = "Images for form: " + jrFormId;
-            albumToUse = client.executeInsert(feed, newAlbum);
+        //creating a new folder
+        com.google.api.services.drive.model.File fileMetadata = new
+                com.google.api.services.drive.model.File();
+        fileMetadata.setName(folderName);
+        fileMetadata.setMimeType("application/vnd.google-apps.folder");
+        if (parentId != null) {
+            fileMetadata.setParents(Collections.singletonList(parentId));
         }
-        return albumToUse;
+
+        com.google.api.services.drive.model.File folder;
+        folder = mDriveService.files().create(fileMetadata)
+                .setFields("id")
+                .execute();
+
+        //adding the permissions to folder
+        Permission sharePermission = new Permission()
+                .setType("anyone")
+                .setRole("reader");
+
+        mDriveService.permissions().create(folder.getId(), sharePermission)
+                .setFields("id")
+                .execute();
+
+        return folder;
     }
 
+    private FileList getFilesFromDrive(String folderName,
+                                       ArrayList<com.google.api.services.drive.model.File> files,
+                                       String parentId) throws IOException {
+        FileList fileList;
+        String pageToken;
+        do {
+            if (parentId == null) {
+                fileList = mDriveService.files().list()
+                        .setQ("name = '" + folderName + "' and " +
+                                "mimeType = 'application/vnd.google-apps.folder'" +
+                                " and trashed=false")
+                        .execute();
+            } else {
+                fileList = mDriveService.files().list()
+                        .setQ("name = '" + folderName + "' and " +
+                                "mimeType = 'application/vnd.google-apps.folder'" +
+                                " and '" + parentId + "' in parents" +
+                                " and trashed=false")
+                        .execute();
+            }
+            for (com.google.api.services.drive.model.File file : fileList.getFiles()) {
+                files.add(file);
+            }
+            pageToken = fileList.getNextPageToken();
```

Comment:
In current scenario, if the user has a large number of files, then it iterates over all the pages and displays all the file. Are we sure that we want to remove this?
Moreover, it is not affecting the performance if the number of files are less.